src/Form/FormMapper.php

```diff
@@ -129,43 +109,36 @@ public function add($name, ?string $type = null, array $options = [], array $fie
         }
 
         $fieldDescription = $this->getAdmin()->createFieldDescription(
-            $name instanceof FormBuilderInterface ? $name->getName() : $name,
+            $name,
             $fieldDescriptionOptions
         );
 
         // Note that the builder var is actually the formContractor:
         $this->builder->fixFieldDescription($fieldDescription);
 
-        if ($fieldName !== $name) {
-            $fieldDescription->setName($fieldName);
```

Comment:
I would say it's better to have a fieldDescription with the name passed to the FormMapper like `foo.bar` instead of the field passed to the formBuilder `foo__bar`.

This way it's consistent with others mapper and for instance I can write
```
$this->getAdmin()->getLabelTranslatorStrategy()->getLabel($fieldDescription->getName(), 'form', 'label');
```